src/ossia/network/base/node_functions.cpp

```diff
@@ -640,6 +642,42 @@ std::vector<ossia::net::node_base*> list_all_children(ossia::net::node_base* nod
   return list;
 }
 
+/**
+ * @brief fuzzysearch: search for nodes that match the pattern string
+ * @param node: root node from where to start
+ * @param pattern: strings to search
+ * @return a vector of pairs of matching score and node pointer sorted in descending order
+ */
+std::vector<std::tuple<double, std::string, ossia::net::node_base*>>
+fuzzysearch(ossia::net::node_base* node, const std::vector<std::string>& patterns)
```

Comment:
right now it looks good, but at some point we may want to have a 

```C++
struct fuzzy_search_state {
    std::vector<std::string> oscnames;
   std::vector<score> scores;
};
```

which would be passed to fuzzysearch, so that we can keep the memory allocations around... as right now it'll allocate a lot for big patches.

Reply:
fixed with d9c9c748